src/grpc/stream_infer_handler.cc

```diff
@@ -140,6 +140,17 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
   // This means that we only need to take care of
   // synchronizing this thread and the ResponseComplete
   // threads.
+  // We need an explicit finish indicator. Can't use 'state->step_'
+  // because we launch an async thread that could update 'state's
+  // step_ to be FINISH before this thread exits this function.
+  bool finished = false;
+  if (state->context_->grpc_strict_) {
+    std::lock_guard<std::recursive_mutex> lock(state->context_->mu_);
+    // Check if stream error detected and already connection ended
+    if (state->context_->IsGRPCStrictError()) {
+      return finished;
```

Comment:
Do you need to set `state->step_ = Steps::FINISH;` here for any dependent cleanup logic?

Reply:
It looks like it would affect the flow in the main loop https://github.com/triton-inference-server/server/blob/9ad856c4f67e62226eff40d757e6181ddf97c9a2/src/grpc/stream_infer_handler.cc#L149 and whats returned by this HandleCompletion func: https://github.com/triton-inference-server/server/blob/9ad856c4f67e62226eff40d757e6181ddf97c9a2/src/grpc/infer_handler.h#L693, possibly putting it back on completion queue